pkg/providers/nats/source/model_source.go

```diff
@@ -0,0 +1,304 @@
+package source
```

Comment:
i think this file better be named not as model_source, but simply - source.
other connectors contains model_* for plain structs, that's only contains configuration data and no active feature.